File: eRecipes/eRecipes.Service/Database/Recept.cs
using System;
using System.Collections.Generic;

namespace eRecipes.Service.Database;

public partial class Recept
{
    public int ReceptId { get; set; }

    public string Naziv { get; set; } = null!;

    public string? OpisRecepta { get; set; }

    public byte[]? Slika { get; set; }

    public int? VrijemePripreme { get; set; }

    public int? KorisnikId { get; set; }

    public DateTime? DatumObjave { get; set; }

    public bool? Premium { get; set; }

    public int? VrstaJelaId { get; set; }

    public int? KategorijaId { get; set; }

    public bool? Status { get; set; }

    public string? StateMachine { get; set; }

    public virtual ICollection<Izvjestaj> Izvjestajs { get; set; } = new List<Izvjestaj>();

    public virtual Kategorija? Kategorija { get; set; }

    public virtual Korisnik? Korisnik { get; set; }

    public virtual ICollection<Lajkovi> Lajkovis { get; set; } = new List<Lajkovi>();

    public virtual ICollection<OmiljeniRecept> OmiljeniRecepts { get; set; } = new List<OmiljeniRecept>();

    public virtual ICollection<ReceptSastojak> ReceptSastojaks { get; set; } = new List<ReceptSastojak>();

    public virtual VrstaJela? VrstaJela { get; set; }
}

File: eRecipes/UI/erecipes_mobile/lib/models/recept.dart


import 'package:erecipes_mobile/models/korisnik.dart';
import 'package:erecipes_mobile/models/recept_sastojak.dart';
import 'package:json_annotation/json_annotation.dart';

part 'recept.g.dart';

@JsonSerializable()
class Recept{
  int? receptId;
  String? naziv; 
  String? slika;
  int? vrstaJelaId;
  int? kategorijaId;
  int? vrijemePripreme;
  int? korisnikId;
  bool? premium;
  String? stateMachine;
  String? opisRecepta;
  DateTime? datumObjave;
  String? opisPripreme;
  List<ReceptSastojak>? sastojci;
  Korisnik? korisnik;
  bool? isFavorite;
  List<String>? sastojcii;
   Recept({
    required this.naziv,
    required this.opisRecepta,
    required this.opisPripreme,
    required this.vrijemePripreme,
    this.kategorijaId,
    this.vrstaJelaId,
    this.slika,
  });

  factory Recept.FromJson(Map<String,dynamic> json)=> _$ReceptFromJson(json);

  Map<String,dynamic> toJson() => _$ReceptToJson(this);
}